File: src/code/code.js
import { getRevenue, getPrice, getProfit } from "getFinanceData";
import { getCity, getCountry, getStreet } from "getUserAddress";

export function transformEvent(event, metadata) {
  return {
    revenue: getRevenue(event.properties || {}),
    price: getPrice(event.properties || {}),
    profit: getProfit(event.properties || {}),
    city: getCity(event.context?.traits?.address || {}),
    country: getCountry(event.context?.traits?.address || {}),
    street: getStreet(event.context?.traits?.address || {}),
  };
}

File: src/code/code.py
from getFinanceData14 import getRevenue, getPrice, getProfit
from getUserAddress14 import getCity, getCountry, getStreet

def transformEvent(events,metadata):
    return {
        'revenue': getRevenue(events.get('properties')),
        'price': getPrice(events.get('properties')),
        'profit': getProfit(events.get('properties')),
        'city': getCity(events.get('context', {}).get('traits', {}).get('address')),
        'country': getCountry(events.get('context', {}).get('traits', {}).get('address')),
        'street': getStreet(events.get('context', {}).get('traits', {}).get('address'))
    }
